library/src/updater.rs

```diff
@@ -161,18 +177,36 @@ pub fn init(
         return Err(InitError::AlreadyInitialized);
     }
 
-    let _ = with_config(|config| {
-        UpdaterState::load_or_new_on_error(
+    handle_prior_boot_failure_if_necessary()
+}
+
+/// If, at initialization time, we detect that we were in the process of booting a patch, report a
+/// failure to boot for that patch and queue an event to report the failure.
+pub fn handle_prior_boot_failure_if_necessary() -> Result<(), InitError> {
+    with_config(|config| {
```

Comment:
Should this be using your with_state?

Reply:
That'd be nice, but we need the `config` to create the PatchEvent later on